src/DevCenter/DevCenterData.AutoRest/README.md

```diff
@@ -0,0 +1,147 @@
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/devboxes/{devBoxName}:stop"].post.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/devboxes/{devBoxName}:restart"].post.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/environments/{environmentName}"].put.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "#/definitions/Environment"}
+      }
+  - from: swagger-document
+    where: $.paths["/projects/{projectName}/users/{userId}/environments/{environmentName}"].delete.responses
+    transform: >
+      $['200'] = {
+        "description": "OK. The request has succeeded.",
+        "schema": {"$ref": "devcenter.json#/definitions/OperationStatus"}
+      }
+  - from: swagger-document
+    where: $.paths["/devboxes"].get.operationId
+    transform: return "DevBoxes_ListAllDevBoxes"
+  - from: swagger-document
+    where: $.paths["/users/{userId}/devboxes"].get.operationId
+    transform: return "DevBoxes_ListAllDevBoxesByUser"
+  - where:
+      subject: ^(.*)(DevBoxPool)(.*)$
+    set:
+      subject: Pool
+  - where:
+      subject: ^(.*)(DevBoxSchedule)(.*)$
+    set:
+      subject: Schedule
+  - where:
+      subject: ^(.*)(DevCenterProject)(.*)$
+    set:
+      subject: Project
+  - where:
+      subject: ^(.*)(EnvironmentCatalog)(.*)$
+    set:
+      subject: Catalog
+# Matches any subject that is not strictly "DevBox" or "Environment" (eg. still includes DevBoxAction)
+  - where:
+      subject: ^(?!DevBox$|Environment$).*
+      parameter-name: UserId
+    set:
+      default:
+        script: '"me"'
+# Matches cmdlets with exact subject DevBox or Environment, but not with verb Get
+  - where:
+      verb: ^(?!Get$)
+      subject: ^(DevBox|Environment)$
+      parameter-name: UserId
+    set:
+      default:
+        script: '"me"'
+  - where:
+      subject: ^(.*)
+      parameter-name: Top
+    hide: true
+  - where:
+      subject: ^(.*)
+      parameter-name: Filter
+    hide: true
+  - where:
+      verb: New
+      variant: ^Create$|^CreateViaIdentity$
+    remove: true
+  - from: swagger-document
+    where: $.definitions.EnvironmentUpdateProperties.properties.parameters
+    transform: $["additionalProperties"] = true
+  - where:
+      subject: ^(.*)
+    hide: true
```

Comment:
Is this a forever choice to hide all cmdlets?

Reply:
Yes this was a choice to hide all cmdlets. For our all our data plane cmdlets we have parameters "DevCenter" and "Endpoint" which we process using ARG and we want to have validation for.